plugins/modules/ec2_vpc_vgw.py

```diff
@@ -325,95 +344,92 @@ def find_vgw(client, module, vpn_gateway_id=None):
         if module.params.get("state") == "present":
             params["Filters"].append({"Name": "state", "Values": ["pending", "available"]})
     try:
-        response = client.describe_vpn_gateways(aws_retry=True, **params)
-    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
-        module.fail_json_aws(e, msg="Failed to describe gateway using filters")
+        response = describe_vpn_gateways(client, **params)
+    except AnsibleEC2Error as e:
+        module.fail_json_aws_error(e)
 
-    return sorted(response["VpnGateways"], key=lambda k: k["VpnGatewayId"])
+    return sorted(response, key=lambda k: k["VpnGatewayId"])
 
 
-def ensure_vgw_present(client, module):
-    # If an existing vgw name and type matches our args, then a match is considered to have been
-    # found and we will not create another vgw.
+def ensure_vgw_present(client, module: AnsibleAWSModule) -> Tuple[bool, Dict[str, Any]]:
+    changed = False
+    params = {
+        "Name": module.params.get("name"),
+        "VpcId": module.params.get("vpc_id"),
+        "VpnGatewayIds": module.params.get("vpn_gateway_id"),
+        "Tags": module.params.get("tags") or {},
+        "PurgeTags": module.params.get("purge_tags", False),
```

Comment:
Also, I don't understand why params is needed, since you only used params["VpcId"] in handle_existing_vgw() and create_new_gw(), which can be accessed from module.params, or am I wrong?

Reply:
right. removed.